tests/integration/components/impagination-dataset-test.js

```diff
@@ -96,6 +96,17 @@ describe('Integration | Component | ImpaginationDataset', function() {
       expect(first.content).to.be.null;
     });
 
+    describe("when the dataset is recomputed", function() {
```

Comment:
I would prefer to not reset the `init` hook and use the `sinon.calledTwice` function.
Also in the `above` describe block (line 81 of this file), make sure that `init.calledOnce` evaluates to true, much like `observe.calledOnce`.

```    javascript
describe("when the dataset is recomputed", function() {
  beforeEach(function() {
    this.set('page-size', 100);
  });
  
  it("fires the on-init hook again", function() {
    expect(init.calledTwice).to.be.true;
  });
});
```